src/iccpd/src/iccp_ifm.c

```diff
@@ -1016,17 +1034,42 @@ void iccp_parse_if_vlan_info_from_netlink(struct nlmsghdr *n)
 
                     vinfo = RTA_DATA(i);
 
+                    ICCPD_LOG_DEBUG(__FUNCTION__, "vlan %d, flag %x, lif_name %s", vinfo->vid, vinfo->flags, lif->name);
+                    if (vinfo->flags & BRIDGE_VLAN_INFO_RANGE_BEGIN) {
+                        start_vid = vinfo->vid;
+                        tmp_vid = start_vid + 1;
+                    }
+
+                    if (vinfo->flags & BRIDGE_VLAN_INFO_RANGE_END) {
+                        end_vid = vinfo->vid;
+
+                        if ((start_vid != 0) && (end_vid > tmp_vid))
+                        {
+                            while (tmp_vid < end_vid)
+                            {
+                                local_if_add_vlan(lif, tmp_vid);
+                                tmp_vid++;
+                            }
+                            tmp_vid = 0; start_vid = 0; end_vid = 0;
+                        }
+                    }
+
                     local_if_add_vlan(lif, vinfo->vid);
                 }
 
                 /*After update vlan list, remove unused item*/
-                LIST_FOREACH(vlan, &(lif->vlan_list), port_next)
+                RB_FOREACH_SAFE (vlan, vlan_rb_tree, &(lif->vlan_tree), vlan_temp)
                 {
                     if (vlan->vlan_removed == 1)
                     {
                         ICCPD_LOG_DEBUG(__FUNCTION__, "Remove %s from VLAN %d", lif->name, vlan->vid);
-
-                        LIST_REMOVE(vlan, port_next);
+#if 0
```

Comment:
Are these code changes needed? 
All the code in this function has been marked under "#if 0"

Reply:
@prvattem: Thanks for the comment. The unused function iccp_parse_if_vlan_info_from_netlink is being removed.